stbt-batch.d/run

```diff
@@ -1,315 +1,181 @@
-  [[ $keep_going -gt 1 ]]
-}
-
-template() {
-  local template=$(
-    sed -n 's,^.*stbt-run: Searching for \(.*\.png\)$,\1,p' stderr.log |
-    tail -1)
-  [ -f "$template" ] && cp "$template" template.png
-}
-
-backtrace() {
-  local gdbcommand corefile=$1
-  gdbcommand=$(mktemp -t report.XXX) || die "Failed to create temp file"
-  echo "thread apply all bt" > $gdbcommand
-  gdb $(which python) $corefile -batch -x $gdbcommand &> backtrace.log
-  rm -f $gdbcommand
-}
-
-user_command() {
-  local c=$("$runner"/../stbt-config batch.$1 2>/dev/null)
-  [[ -z "$c" ]] && return
-  "$c" $2 </dev/null
-}
-
-check_capture_hardware() {
-  case "$("$runner"/../stbt-config global.source_pipeline | awk '{print $1}')" in
-    v4l2src)
-      if grep -q "Cannot identify device '/dev/v" failure-reason; then
-        echo "v4l2 device not found; exiting."
+import argparse
+import os
+import signal
+import subprocess
+import sys
+from distutils.spawn import find_executable
+
+
+def main(argv):
+    runner = os.path.dirname(os.path.abspath(__file__))
+
+    parser = argparse.ArgumentParser(usage=(
+        "stbt batch run [options] test.py [test.py ...]\n"
+        "stbt batch run [options] test.py arg [arg ...] -- "
+        "test.py arg [arg ...] [-- ...])"))
+    parser.add_argument(
+        '-1', '--run-once', action="store_true", help=(
+            'Run once. The default behaviour is to run the test repeatedly as '
+            'long as it passes.'))
+    parser.add_argument(
+        '-k', '--keep-going', action="count", help=(
+            'Continue running after failures.  Provide this argument once to '
+            'continue running after "uninteresting" failures, and twice to '
+            'continue running after any failure (except those that would '
+            'prevent any further test from passing).'))
+    parser.add_argument(
+        '-d', '--debug', action="store_true", help=(
+            'Enable "stbt-debug" dump of intermediate images.'))
+    parser.add_argument(
+        '-v', '--verbose', action="count", default=0, help=(
+            'Verbose. Provide this argument once to print stbt standard '
+            'output. Provide this argument twice to also print stbt stderr '
+            'output'))
+    parser.add_argument(
+        '-o', '--output', default=os.curdir, help=(
+            'Output directory to save the report and test-run logs under '
+            '(defaults to the current directory).'))
+    parser.add_argument(
+        '-t', '--tag', help=(
+            'Tag to add to test run directory names (useful to differentiate '
+            'directories when you intend to merge test results from multiple '
+            'machines).'))
+    parser.add_argument('test_name', nargs=argparse.REMAINDER)
+    args = parser.parse_args(argv[1:])
+
+    if args.tag is not None:
+        tag = '-' + args.tag
+    else:
+        tag = ""
+
+    os.environ['PYTHONUNBUFFERED'] = 'x'
+
+    term_count = [0]
```

Comment:
Fixed.  I've removed `stop` from `on_term` because in future commits I will be simplifying the loop and that will let me remove `stop`.